helper/helper.go

```diff
@@ -0,0 +1,81 @@
+package helper
+
+import (
+	"archive/tar"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func GetApplication(applicationsPath, applicationUUID string) (string, string, error) {
+	commit, err := getCommit(applicationsPath, applicationUUID)
+	if err != nil {
+		log.Printf("Failed to get commit. Error: %s\r\n", err)
+		return "", "", err
+	}
+
+	basePath := filepath.Join(applicationsPath, applicationUUID, commit)
+	tarball := filepath.Join(basePath, "binary.tar")
+	application := filepath.Join(basePath, "application.zip")
+
+	_, err = os.Stat(application)
+	if err != nil {
```

Comment:
If the error _isn't_ `nil` you proceed? I'm confused here - is this because you expect the application to exist otherwise you untar?

In that case you should emphatically _not_ ignore the error type and assume it means it doesn't exist. You should use the [os.IsNotExist](https://golang.org/pkg/os/#IsNotExist) function to determine if the error is of that type and if not return the error.

Also again you're creating unnecessary nesting. If you combine the use of declaring vars at the top (see top comment for this file) and a guard clause here this can be made a lot cleaner.